src/StringViews.jl

```diff
@@ -86,6 +86,8 @@ Base.:(==)(s1::StringViewAndSub, s2::StringAndSub) = s2 == s1
 
 Base.typemin(::Type{StringView{Vector{UInt8}}}) = StringView(Vector{UInt8}(undef,0))
 Base.typemin(::T) where {T<:StringView} = typemin(T)
+Base.one(::Union{T,Type{T}}) where {T<:StringView} = typemin(T)
+Base.oneunit(::Union{T,Type{T}}) where {T<:StringView} = typemin(T)
```

Comment:
There is a default definition `oneunit(x::T) where {T} = T(one(x))` in Base, shouldn't that suffice?

Reply:
Here's what I get with only `one` defined:

```julia
julia> using StringViews

julia> sv = StringView([0x65])
"e"

julia> Base.one(::Union{T,Type{T}}) where {T<:StringView} = typemin(T)

julia> one(sv)
""

julia> oneunit(sv)
MethodError: Cannot `convert` an object of type StringView{Vector{UInt8}} to an object of type Vector{UInt8}
```

Reply:
I think that's because there is currently no `StringView(::StringView)` constructor.  If that is added then it should work.

Reply:
There does appear to be a [Stringview(::StringView) constructor](https://github.com/mnemnion/StringViews.jl/blob/free-monoid-identity/src/StringViews.jl#L41). Adding a `StringView(::Type{StringView})` constructor might make Base `oneunit` work, it might not, I haven't tried it, but is that actually an operation StringViews should support? There isn't a `String(::Type{String})`, just the first thing I thought to try.

Reply:
So `one(StringView)` doesn't work either, because `typemin` is only defined for `Type{StringView{Vector{UInt8}}}`.  That seems also worth addressing. But it isn't the only thing wrong with the patch. 

For instance, `one(StringView(""))` fails with `MethodError: no method matching one(::Type{Base.CodeUnits{UInt8, String}})`.  `typemin` is also not defined for that type.  It should be, it _really_ should be, but that doesn't help us here. 

I'm going to need to take some time to work out the missing pieces here, figure out if `CodeUnits` is the only sharp edge in Base at least, and figure out how to cover the bases, at least until the next guy forgets to define typemin for an `<:AbstractVector{T}` because Julia can't define interfaces, this is fine, everything is fine, really, just gotta, implement, some things.

Reply:
Do you think [this definition](https://github.com/mnemnion/StringViews.jl/blob/free-monoid-identity/src/StringViews.jl#L88) is intentional, btw? 

```julia
Base.typemin(::StringView{S}) where {S} = StringView(typemin(S))
```
Makes more sense to me, or with `Union{Type{StringView{S},StringView{S}}` etc. With that, and one or more special cases for missing `typemin` and this should be good to go.

Reply:
I didn't add `Base.typemin(::StringView{S}) where {S}`, and it seems to me that a `typemin(::Type{StringView})`, without a parameter, is underdefined and should fail. I can revisit those decisions if desired, meanwhile I'm going to mark this thread resolved.

Reply:
There is a `StringView(::StringView)`, but not a `StringView{T}(::StringView{T})` constructor.